google/cloud/aiplatform/datasets/tabular_dataset.py

```diff
@@ -0,0 +1,149 @@
+                The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            gcs_source: Optional[Sequence[str]] = None:
+                Google Cloud Storage URI(-s) to the
+                input file(s). May contain wildcards. For more
+                information on wildcards, see
+                https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames.
+            bq_source: Optional[str] = None:
+                BigQuery URI to the input table.
+            labels: Optional[Dict] = {}
+                The labels with user-defined metadata to organize your
+                Datasets.
+
+                Label keys and values can be no longer than 64 characters
+                (Unicode codepoints), can only contain lowercase letters,
+                numeric characters, underscores and dashes. International
+                characters are allowed. No more than 64 user labels can be
+                associated with one Dataset (System labels are excluded).
+
+                See https://goo.gl/xmQnxf for more information and examples
+                of labels. System reserved label keys are prefixed with
+                "aiplatform.googleapis.com/" and are immutable.
+            project: Optional[str] = None,
+                Project to upload this model to. Overrides project set in
+                aiplatform.init.
+            location: Optional[str] = None,
+                Location to upload this model to. Overrides location set in
+                aiplatform.init.
+            credentials: Optional[auth_credentials.Credentials] = None,
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+            request_metadata: Optional[Sequence[Tuple[str, str]]] = ()
+                Strings which should be sent along with the request as metadata.
+            sync: (bool) = True
+                Whether to execute this method synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+
+        Returns:
+            tabular_dataset: TabularDataset
+                Instantiated representation of the managed tabular dataset resource.
+
+        """
+
+        utils.validate_display_name(display_name)
+        cls.metadata_schema_uri = schema.dataset.metadata.tabular
+        datasource = TabularDatasource(gcs_source, bq_source)
+
+        return cls._create_and_import(
+            display_name=display_name,
+            metadata_schema_uri=cls.metadata_schema_uri,
+            dataset_metadata=datasource.dataset_metadata,
+            labels=labels,
+            project=project,
+            location=location,
+            credentials=credentials,
+            request_metadata=request_metadata,
+            sync=sync,
+        )
+
+    def import_data(self):
```

Comment:
Good catch.

I wish there was a cleaner way to do this, like split out `import_data` (and other import functionality) from Dataset so that only relevant subclasses of ImportableDataset would even have this function. 

Perhaps we can do a separate refactor:
1. DatasetBase only includes Dataset creation functionality
2. Dataset inherits DatasetBase and adds import functionality
3. ImageDataset would inherit Dataset
4. TabularDataset would inherit DatasetBase

Reply:
Seems big enough to be a separate PR.